builtin/aws/ecs/platform.go

```diff
@@ -1102,23 +1411,26 @@ func (p *Platform) Launch(
 		Tags: []*ecs.Tag{
 			{
 				Key:   aws.String("waypoint-app"),
-				Value: aws.String(app.App),
+				Value: aws.String(src.App),
 			},
 		},
 	}
 
-	if taskRoleArn != "" {
-		registerTaskDefinitionInput.SetTaskRoleArn(taskRoleArn)
+	if taskRole != nil && taskRole.Arn != "" {
+		registerTaskDefinitionInput.SetTaskRoleArn(taskRole.Arn)
 	}
 
-	var taskOut *ecs.RegisterTaskDefinitionOutput
+	ecsSvc := ecs.New(sess)
 
+	var taskOut *ecs.RegisterTaskDefinitionOutput
+	var err error
 	// AWS is eventually consistent so even though we probably created the resources that
 	// are referenced by the task definition, it can error out if we try to reference those resources
 	// too quickly. So we're forced to guard actions which reference other AWS services
 	// with loops like this.
-	for i := 0; i < 30; i++ {
-		taskOut, err = ecsSvc.RegisterTaskDefinition(&registerTaskDefinitionInput)
+
+	for i := 0; i <= awsCreateRetries; i++ {
```

Comment:
Longer term it probably makes sense to have some kind of exponential backoff with a context cancel.